### ISSUE TITLE
[Feature Request]: channel audio mode

### ISSUE THREAD

**Issue by greg321321** [tubearchivist/tubearchivist#623]
### Your Feature Request

Provide an audio mode to channels.

User story #1:
1. As a user, I want to play tube archivist channel content as audio only on my podcast app.

Expected Behavior #1:
1. TA shall provide an "audio/video podcast RSS link" on the TA channel page.
2. TA shall provide an "audio/video podcast RSS link" on the TA playlist page.
3. Upon request of the "audio/video podcast RSS link" data, TA shall provide podcast data in accordance with the [PSP-1: The Podcast RSS Standard](https://github.com/Podcast-Standards-Project/PSP-1-Podcast-RSS-Specification).
4. Upon request for audio media, TA shall provide podcast audio data utilizing ffmpeg on the fly transcoding.
5. TA shall provide podcast audio data in mp3 format.
6. TA podcasts shall be consumable by the following applications: [Antenna Pod](https://www.pcmag.com/picks/the-best-podcast-player-apps).
7. TA shall protect podcast data from unauthorized access.
8. When a podcast app sends more than one simultaneous request for podcast content, TA shall prevent simultaneous requests from negatively impacting the user experience on the main GUI.

Guidance per Simon bbilly:
Step one is to have audio mode, that should be functional in the TA interface, so you can listen to audio only. This would have to be configured per channel, ideally the options would be: video only (currently), audio only, video + audio. This would give all options. Then audio only should download the mp3 directly with yt-dlp, and video + audio could use ffmpeg for post-processing to extract the audio from the video file. 

### Your help is needed!

- [X] Yes I will work on this in the next few days or weeks.

**Comment by bbilly1** [comment]
There are many popular podcasts out there that are pretty much talk shows, they have an interview that is filmed, but they are also a podcast, as in you don't really need the video to follow the conversation, but you might want to. So you might want to get the audio version in your podcast app, but also have the video version to watch the interview. I've also learned that YT has a separate /podcasts section:

Examples:
https://www.youtube.com/@JupiterBroadcasting/podcasts
https://www.youtube.com/@LinusTechTips/podcasts
https://www.youtube.com/@H3Podcast/podcasts
https://www.youtube.com/@ThroughTheWeb/podcasts

But some research also showed, that most podcasts are playlists and not channels, or group of playlists in a channel. So the previous thinking of having audio channels is still valid for music channels, but not really for podcasts and talkshows.

Audio download on a separate device shouldn't be handled in this application, there are a lot of great podcast apps that can take an RSS feed, we just need to parse the metadata correctly, feed specs is public, we should be able to produce this as expected then users can pick their favorite podcast app and TA is just the server. Audio file can just by `<youtube-id>.mp3` as the feed will contain the metadata like title etc., so we don't need to burn our hands again with producing valid file names from an arbitrary unicode string.

Deleting should probably not happen automatically like that, might be better to have a `delete audio files`, `delete video files` functionality or something like that that's separate from that. As an archival project after all...

All in all, that's not fully thought through...

**Comment by greg321321** [comment]
I issued a pull request for an initial proof of concept. Works ok with two Android podcast apps. Needs more testing and some code polishing.

**Comment by greg321321** [comment]
Screenshot of PodcastAddict playing TA channel content.

![Screenshot_20231209-204150](https://github.com/tubearchivist/tubearchivist/assets/134153833/e92f5e01-9c6e-4ac1-8e65-5ba51aff6193)

**Comment by greg321321** [comment]
![Screenshot from 2023-12-10 08-39-04](https://github.com/tubearchivist/tubearchivist/assets/134153833/d9d6d8ad-ee9a-45e0-a948-2eacf828df78)
![Screenshot from 2023-12-10 08-39-16](https://github.com/tubearchivist/tubearchivist/assets/134153833/94cf7d55-6c1f-488b-b7b5-53924375790b)

**Comment by greg321321** [comment]
pull request ready for review

**Comment by circle3451** [comment]
Wouldnt it either way be nice to have the audio only option for music as well?

**Comment by greg321321** [comment]
makes sense. I implemented both an audio and video RSS feed. Just waiting on admins to approve merge request so it can be tested further.

**Comment by greg321321** [comment]
Now that custom playlist are on master branch, I added podcast links on playlists as well.

**Comment by Dontkickmi22** [comment]
Second to audio mode.

**Comment by TemplarB** [comment]
I second the idea. Currently, I mostly use this software to get audiobooks (legal, of old, out of copyright short stories, mostly SF), so I set "bestaudio" in download format, and then in channels I want video, I'd like to have it as a simple button/download option
